apps/webcert/src/page/PPRegistration/PPRegistrationStep02.tsx

```diff
@@ -19,23 +18,66 @@ export function PPRegistrationStep02() {
   const navigate = useNavigate()
 
   const [showPasteError, setShowPasteError] = useState(false)
+  const [previousZipCode, setPreviousZipCode] = useState<string>('')
+  const [clearZipCodeInfo, setClearZipCodeInfo] = useState(false)
   const { phoneNumber, email, emailRepeat, address, zipCode, city, county, municipality } = useAppSelector(
     (state) => state.ui.pp.step02.data,
     isEqual
   )
   const errors = useAppSelector((state) => state.ui.pp.step02.errors)
-  const [getZipCodeInfo, { data: zipCodeInfo, isError: isZipCodeError, isSuccess: isZipCodeSuccess }] = useLazyGetZipCodeInfoQuery()
+  const [getZipCodeInfo, { data: zipCodeInfo, isError: isZipCodeError }] = useLazyGetZipCodeInfoQuery()
+
+  const activeZipCodeInfo = clearZipCodeInfo ? undefined : zipCodeInfo
 
   useEffect(() => {
-    if (`${zipCode}`.length === 5 && zipCodeInfo?.at(0)?.zipCode !== zipCode) {
+    if (`${zipCode}`.length === 5 && activeZipCodeInfo?.at(0)?.zipCode !== zipCode) {
+      const zipCodeChanged = previousZipCode !== zipCode && previousZipCode !== ''
+      if (zipCodeChanged) {
+        dispatch(updateField({ field: 'city', value: '' }))
+        dispatch(updateField({ field: 'municipality', value: '' }))
+        dispatch(updateField({ field: 'county', value: '' }))
+      }
```

Comment:
This could possibly be handled in `apps/webcert/src/store/pp/ppStep02ReducerSlice.ts` with a 

```ts
builder.addMatcher(api.endpoints.getZipCodeInfo.matchPending, (state, { payload }) => {
  state.data.city = ''
  state.data.county = ''
  state.data.municipality = ''
})
```